### PR TITLE
Recover deprecated property in Swagger 2.0 support

### LINKED ISSUE #646: unable to declare endpoints as deprecated in swagger
As a developer, I need the ability to deprecate endpoints in swagger. For instance, see the deprecated `/pet/findByTags` endpoint in [petstore swagger](http://petstore.swagger.io/v2/swagger.json).

not sure why not working: [deprecate method](https://github.com/scalatra/scalatra/blob/2.5.x/swagger/src/main/scala/org/scalatra/swagger/SwaggerSupport.scala#L251)

Scalatra endpoint:

` apiOperation[Object]("filter")
      deprecate
      summary ("filter something")
      parameter pathParam[String]("field").description("The field to filter on").required
      responseMessages(
      ResponseMessage(code = 401, message = "returns 401 (Unauthorized) if no api-key", responseModel = None)
    )
`

yet `deprecated` does not show up as property in swagger, 
`
  "/filter/{field}": {
      "get": {
        "operationId": "filter",
        "summary": "filter something",
        "parameters": [
          {
            "name": "field",
            "description": "The field to filter on",
            "required": true,
            "in": "path",
            "type": "string"
           }]
`

### PR DESCRIPTION
**Pull request by takezoe**
fixes #646

### CODE REVIEW

File: swagger/src/test/resources/swagger.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "/store/order": {
       "post": {
         "operationId": "placeOrder",
+        "deprecated": false,
```
Comment (jb08) [inline swagger/src/test/resources/swagger.json]: swagger-ui should work without these `false` declarations. See [petstore swagger.json](http://petstore.swagger.io/v2/swagger.json) as an example.

**Comment by jb08** [comment]
swagger-ui should work without the `false` declarations in `swagger.json`. See [petstore swagger.json](http://petstore.swagger.io/v2/swagger.json) as an example.

File: swagger/src/test/resources/swagger.json (hunk 1 above)
Comment (takezoe) [inline swagger/src/test/resources/swagger.json]: Yes, but this makes implementation easier.

Comment (jb08) [inline swagger/src/test/resources/swagger.json]: I verified this PR works well. nice job!

**Comment by jb08** [comment]
this branch can be merged imo

**Pull request merged**